platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,204 @@
+      }
+    }
+  }
+
+  public void init() {
+    checkForMissedServletContexts();
+  }
+
+  @SuppressWarnings("unchecked")
+  private void checkForMissedServletContexts() {
+
+    Optional<BundleContext> optionalBundleContext = getContext();
+    if (!optionalBundleContext.isPresent()) {
+      LOGGER.error(
+          "Problem retrieving the bundle `Platform :: Error :: Page Injector` after it's just been initialized");
+      return; // This should never happen
+    }
+
+    //  Get all registered servletContexts in OSGI
+    final BundleContext context = optionalBundleContext.get();
+    ServiceReference<ServletContext>[] references = null;
+
+    try {
+      references =
+          (ServiceReference<ServletContext>[])
+              context.getAllServiceReferences(
+                  null, "(" + Constants.OBJECTCLASS + "=javax.servlet.ServletContext)");
+
+    } catch (InvalidSyntaxException e) {
+      LOGGER.error(
+          "Problem getting ServletContexts from OSGI, One of the servlets running might print stack traces to the browser. A system restart is recommended. See debug logs for additional details.");
+      LOGGER.debug("Additional Details:", e);
+    }
+
+    if (references == null) return;
+
+    //  Now since we have the list of references loop through them and check to see if they have
+    // error pages added
+    for (ServiceReference<ServletContext> reference : references) {
+
+      final Bundle refBundle = reference.getBundle();
+      final BundleContext bundlectx = refBundle.getBundleContext();
+      ServletContext service = bundlectx.getService(reference);
+
+      //  Need to use the service to get the errorPageErrorHandler through the
+      // ServletContextHandler
+      Optional<ServletContextHandler> optionalHttpContext = getHTTPContext(service, refBundle);
+
+      if (!optionalHttpContext.isPresent()) return;
+
+      ServletContextHandler httpContext = optionalHttpContext.get();
+      ErrorPageErrorHandler errorPageErrorHandler =
+          (ErrorPageErrorHandler) httpContext.getErrorHandler();
+
+      final Map<String, String> errorPages = errorPageErrorHandler.getErrorPages();
+
+      if (errorPages.isEmpty()) {
+
+        createAndAddErrorPageHandler(httpContext, refBundle);
+      }
+    }
+  }
+
+  private void injectErrorPage(ServletContext context, Bundle refBundle) {
+
+    Optional<ServletContextHandler> optionalHttpServiceContext = getHTTPContext(context, refBundle);
+    optionalHttpServiceContext.ifPresent(p -> createAndAddErrorPageHandler(p, refBundle));
+  }
+
+  private Optional<ServletContextHandler> getHTTPContext(ServletContext context, Bundle refBundle) {
+
+    Field field;
+    try {
+      // this grabs the enclosing instance class, which is actually a private class
+      // this is the only way to do this in Java
+      field = context.getClass().getDeclaredField("this$0");
+      field.setAccessible(true);
+    } catch (NoSuchFieldException e) {
+      LOGGER.warn(
```

Comment:
I feel if it's not really possible then there is a greater error in the log that caused this. I could be very wrong here but I don't think this would help an admin get to the bottom of it.